element_calcium_imaging/imaging.py

```diff
@@ -1544,6 +1544,99 @@ def make(self, key):
             raise NotImplementedError("Unknown/unimplemented method: {}".format(method))
 
 
+@schema
+class ProcessingQualityMetrics(dj.Computed):
+    definition = """
+    -> Segmentation
+    """
+
+    class MaskMetrics(dj.Part):
+        definition = """
+        -> master
+        -> Segmentation.Mask
+        ---
+        mask_area=null: float  # Mask area in square micromete. Stringer & Pachitariu (2019).
+        roundness: float       # Roundness between 0 and 1, closer to 1 the rounder. Tegtmeier et al. (2018).
+        """
+
+    class FluorescenceTraceMetrics(dj.Part):
+        definition = """
+        -> master
+        -> Fluorescence.Trace
+        ---
+        skewness: float   # Stringer & Pachitariu (2019)
+        variance: float   # Stringer & Pachitariu (2019)
+        """
+
+    def make(self, key):
+        from scipy.stats import skew
+
+        (
+            mask_xpixs,
+            mask_ypixs,
+            mask_weights,
+            fluorescence,
+            fluo_channels,
+            mask_ids,
+            mask_npix,
+            px_height,
+            px_width,
+            um_height,
+            um_width,
+        ) = (Segmentation.Mask * scan.ScanInfo.Field * Fluorescence.Trace & key).fetch(
+            "mask_xpix",
+            "mask_ypix",
+            "mask_weights",
+            "fluorescence",
+            "fluo_channel",
+            "mask",
+            "mask_npix",
+            "px_height",
+            "px_width",
+            "um_height",
+            "um_width",
+        )
+
+        self.insert1(key)
+
+        roundnesses = np.empty_like(mask_xpixs)
+        for i, (mask_xpix, mask_ypix, mask_weight) in enumerate(
+            zip(mask_xpixs, mask_ypixs, mask_weights)
+        ):
+            eigen_values = np.linalg.eigvals(
+                np.cov(mask_xpix, mask_ypix, aweights=mask_weight)
+            )
+            roundnesses[i] = eigen_values.mean() / eigen_values.max()
+
+        self.MaskMetrics.insert(
+            dict(**key, mask=mask_id, mask_area=mask_area, roundness=roundness)
+            for mask_id, mask_area, roundness in zip(
+                mask_ids,
+                mask_npix * (um_height / px_height) * (um_width / px_width),
+                roundnesses,
+            )
+        )
+
+        fluorescence = np.stack(fluorescence)
+        self.FluorescenceTraceMetrics.insert(
+            [
```

Comment:
You don't need to create a list before inserting. You can insert directly from the iterable.